api-tests/src/steps/cri-steps.ts

```diff
@@ -336,3 +418,38 @@ When(
     );
   },
 );
+
+function assertNoUnexpectedJarProperties(
+  jarPayload: CriStubResponseJarPayload,
+  expectedNonStandardValues: string[] | undefined = undefined,
+) {
+  for (const key of removeStandardJarValues(Object.keys(jarPayload))) {
+    const expectedValueExists =
+      expectedNonStandardValues === undefined
+        ? false
+        : expectedNonStandardValues.includes(key);
+    assert.equal(
+      expectedValueExists,
+      true,
+      `Non-standard JAR property ${key} with value ${JSON.stringify(jarPayload[key as keyof typeof jarPayload])} sent to CRI but not specified in test`,
+    );
+  }
+}
+
+function removeStandardJarValues(jarValues: string[]): string[] {
+  return jarValues.filter(
+    (v) =>
+      v !== "sub" &&
+      v !== "shared_claims" &&
+      v !== "iss" &&
+      v !== "response_type" &&
+      v !== "client_id" &&
+      v !== "govuk_signin_journey_id" &&
+      v !== "aud" &&
+      v !== "nbf" &&
+      v !== "redirect_uri" &&
+      v !== "state" &&
+      v !== "exp" &&
+      v !== "iat",
```

Comment:
Should this just be an array of allowed values - `jarValues.filter((v) => ALLOWED_JAR_ATTRIBUTES.includes(v))`

Reply:
Good point